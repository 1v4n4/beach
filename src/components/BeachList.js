import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import Carousel from 'react-bootstrap/Carousel';
import { getList } from '../actions/listActions';
import Filter from './Filter';
import '../CSS/beachList.css';

const BeachList = () => {
  const [filter, setFilter] = useState('All');
  const dispatch = useDispatch();

  const list = useSelector((state) => state.list);

  console.log('list', list);
  const FetchBeach = () => {
    dispatch(getList());
  };

  useEffect(() => {
    FetchBeach();
  }, []);

  const showData = () => {
    if (list.loading) {
      return <h1 style={{ marginTop: '25vh' }}>Loading</h1>;
    }

    if (list.data && list.data.length > 0) {
      let beaches = [];
      if (filter === 'All' || filter === 'COUNTIES') { beaches = list.data; } else {
        beaches = list.data.filter((beach) => beach.county === filter);
      }
      return (
        <div className="ing-btn-div">
          <Carousel variant="dark">
            {beaches.map((beach, index) => (
              <Carousel.Item key={beach.id}>
                <img
                  className="d-block w-100"
                  src={beach.photo1}
                  alt={beach.name}
                />
                <Carousel.Caption>
                  <Link className="decoration-none" to={{ pathname: `/beach/${beach.id}`, state: { beachid: `${beach.id}` } }}><h5>{beach.name}</h5></Link>
                  <p>{beach.description}</p>

                </Carousel.Caption>
                {' '}
                <p>
                  {index + 1}
                  /
                  {' '}
                  {beaches.length}
                </p>
              </Carousel.Item>
            ))}
          </Carousel>
        </div>
      );

      //       <div className="beach-card" key={beach.id}>
      //         <Link to={{ pathname: `/beach/${beach.id}`,
      // state: { beachid: `${beach.id}` } }}><h2>{beach.name}</h2></Link>
      //         <h4>{beach.location}</h4>
      //       </div>
      //     ))}
      //   </div>

      // );
    }
    if (list.errorMSG !== '') {
      return <h1>{list.errorMSG}</h1>;
    }
    return <p />;
  };

  return (
    <div>
      <Filter filter={filter} setFilter={setFilter} />
      {showData()}
    </div>
  );
};

export default BeachList;
