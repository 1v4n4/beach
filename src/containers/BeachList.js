import React, { useEffect, useState } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import Carousel from 'react-bootstrap/Carousel';
import { getList } from '../actions/listActions';
import Filter from '../components/Filter';
import '../CSS/beachList.css';

const BeachList = () => {
  const [filter, setFilter] = useState('All');
  const dispatch = useDispatch();

  const list = useSelector((state) => state.list);

  console.log('list', list);
  const FetchBeach = () => {
    dispatch(getList());
  };

  useEffect(() => {
    FetchBeach();
  }, []);

  const showData = () => {
    if (list.loading) {
      return <h1 className="nav-link bg-light text-center text-dark mt-5">Loading</h1>;
    }

    if (list.data && list.data.length > 0) {
      let beaches = [];
      if (filter === 'All' || filter === 'COUNTIES') { beaches = list.data; } else {
        beaches = list.data.filter((beach) => beach.county === filter);
      }
      return (
        <div className="ing-btn-div">
          <Carousel variant="dark">
            {beaches.map((beach, index) => (
              <Carousel.Item key={beach.id}>
                <img
                  className="d-block w-100 carusel-photo"
                  src={beach.photo1}
                  alt={beach.name}
                />
                <Carousel.Caption>
                  <Link to={{ pathname: `/beach/${beach.id}`, state: { beachid: `${beach.id}` } }}><h2>{beach.name}</h2></Link>
                  <p className="fs-5">{beach.description}</p>
                  <p className="fs-4">
                    {index + 1}
                    /
                    {' '}
                    {beaches.length}
                  </p>

                </Carousel.Caption>

              </Carousel.Item>

            ))}
          </Carousel>
        </div>
      );
    }
    if (list.errorMSG !== '') {
      return <h1 className="nav-link bg-light text-center text-dark mt-5">{list.errorMSG}</h1>;
    }
    return <p />;
  };

  return (
    <div>
      <Filter filter={filter} setFilter={setFilter} />
      {showData()}
    </div>
  );
};

export default BeachList;
