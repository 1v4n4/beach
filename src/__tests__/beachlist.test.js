import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import BeachList from '../containers/BeachList';
import createTestStore from './store';

describe('BeachList container', () => {
  let store;
  beforeEach(() => {
    store = createTestStore();
  });

  test('should match with snapshot', () => {
    const jsx = (
      <Provider store={store}>
        <BrowserRouter>
          <BeachList />
        </BrowserRouter>
      </Provider>
    );
    const tree = (() => renderer.create(jsx).toJSON());
    expect(tree).toMatchSnapshot();
  });
});
