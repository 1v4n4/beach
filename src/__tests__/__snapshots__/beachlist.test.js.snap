// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BeachList container should match with snapshot 1`] = `[Function]`;
